worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala

```diff
@@ -94,15 +94,124 @@ class FetchHandler(val conf: CelebornConf, val transportConf: TransportConf)
       case r: ChunkFetchRequest =>
         handleChunkFetchRequest(client, r)
       case r: RpcRequest =>
-        val msg = Message.decode(r.body().nioByteBuffer())
-        handleOpenStream(client, r, msg)
+        // 1. parse transportMessage
+        try {
+          val pbMsg = TransportMessage.fromByteBuffer(r.body().nioByteBuffer())
+          val pbOpenStream = pbMsg.getPayLoad[PbOpenStream]
+          handleOpenStream(client, r.requestId, pbOpenStream, () => r.body().release())
+        } catch {
+          case _: Exception =>
+            // 2. parse old rpc message to keep client compatible
+            logDebug("use legacy format")
+            val msg = Message.decode(r.body().nioByteBuffer())
+            handleLegacyOpenStream(client, r, msg)
+        }
       case unknown: RequestMessage =>
         throw new IllegalArgumentException(s"Unknown message type id: ${unknown.`type`.id}")
     }
   }
 
-  // here are BackLogAnnouncement,OpenStream and OpenStreamWithCredit RPCs to handle
-  def handleOpenStream(client: TransportClient, request: RpcRequest, msg: Message): Unit = {
+  def handleOpenStream(
```

Comment:
I don't think we should add this method, because its structure is mostly the same with `handleLegacyOpenStream`. This is more error-prone because when we want to change the logic, we have to modify both methods.

Instead, IMO we should use one method, and change the signature like
```
  def handleOpenstream(
      client: TransportClient,
      request: RpcRequest,
      shuffleKey: String,
      fileName: String,
      startMapIndex: Int,
      endMapIndex: Int,
      initialCredit: Int,
      startIndex: Int,
      endIndex: Int,
      isLegacy: Boolean)
```